dask/array/core.py

```diff
@@ -2585,15 +2645,41 @@ def store(x, out, index, lock, region):
             if lock:
                 lock.release()
 
-        return None
+        return result
 
     slices = slices_from_chunks(arr.chunks)
 
     name = 'store-%s' % arr.name
     dsk = dict()
     for t, slc in zip(core.flatten(arr.__dask_keys__()), slices):
         store_key = (name,) + t[1:]
-        dsk[store_key] = (store, t, out, slc, lock, region)
+        dsk[store_key] = (store, t, out, slc, lock, region, keep)
+
+    return dsk
+
+
+def retrieve_from_ooc(out_dsk):
+    def load(x, index, lock, region):
+        result = None
+
+        subindex = index
+        if region is not None:
+            subindex = fuse_slice(region, index)
+
+        if lock:
+            lock.acquire()
+        try:
+            result = x[subindex]
+        finally:
+            if lock:
+                lock.release()
+
+        return result
+
+    dsk = dict()
+    for store_key in out_dsk:
+        load_key = ('load-%s' % store_key[0][len("store-"):],) + store_key[1:]
+        dsk[load_key] = (load, store_key,) + out_dsk[store_key][3:-1]
```

Comment:
If possible it would be nice to avoid placing dynamically generated functions (like `load` above) into task graphs.  This tends to be a bottleneck for larger computations on distributed machines where function serialization can become an issue.

Reply:
Hmm...so this is simply mirroring the behavior of `insert_to_ooc` with `store`. If having these functions dynamically generated is an issue, it seems like we should start by fixing `insert_to_ooc`.

Reply:
`da.store` was left derelict for some time.  We just never touched it after the change to the dask.distributed scheduler.  My apologies for the stale code.